public/scripts/setup.js

```diff
@@ -2,8 +2,8 @@ function initializeUser(access_token, refresh_token){
   // checkForExistingHistory();
   var userLastId = $("#lastId").val();
   localStorage.setItem("userLastId", userLastId);
+	// XXX maybe `getTrackForUser()` would be better named as `getLastFmHistory()` or similar?
   getTrackForUser(getFourWeeks(), getYesterday());
-  getPlaylists(access_token);
```

Comment:
this is the main change - instead of immediately calling getPlaylists() as soon as we've fired off our 40 last.fm api requests, we instead don't call it until all of those requests' responses have been received and processed